src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -0,0 +1,247 @@
+import {
+  Alert,
+  Autocomplete,
+  Box,
+  Button,
+  Link,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material"
+import React, { useMemo, useState } from "react"
+import { commify, formatBNToString, isNumberOrEmpty } from "../../utils"
+import ArrowDownIcon from "@mui/icons-material/KeyboardArrowDown"
+import { BigNumber } from "ethers"
+import { Gauge } from "../../utils/gauges"
+import { GaugeController } from "../../../types/ethers-contracts/GaugeController"
+import VoteHistory from "./Votes"
+import { enqueuePromiseToast } from "../../components/Toastify"
+import { useActiveWeb3React } from "../../hooks"
+import { useQuery } from "@tanstack/react-query"
+import { useTranslation } from "react-i18next"
+
+interface OnChainVoteProps {
+  veSdlBalance: BigNumber
+  gauges: Partial<{
+    [lpTokenAddress: string]: Gauge
+  }>
+  gaugeControllerContract: GaugeController
+}
+export type GaugeName = {
+  [gaugeAddress: string]: {
+    address: string
+    gaugeName: string
+  }
+}
+
+export type VotesType = {
+  [gaugeAddress: string]: {
+    gaugeName: string
+    weight: BigNumber
+    voteDate: number
+  }
+}
+
+export default function OnChainVote({
+  veSdlBalance,
+  gauges,
```

Comment:
Is there a specific reason to pass gauges as a prop? It's already accessible through context.

Reply:
Because we still don't have loading status for `gauges` on context, I am checking gauge load status by following on parent component.
```
const isGaugesLoading = Object.keys(gauges).length === 0
```

Will update this logic while update component with `useQuery`